### ISSUE TITLE
Allow constant exports doesn't work with objects.

### ISSUE THREAD

**Issue by AaronPowell96** [ArnaudBarre/eslint-plugin-react-refresh#36]
I am looking to create some way where these constants would work.
If I am able to say these are immutable with as const or if the plugin would be able to infer that simply by a full capitalisation of a variable that would be nice. If I write `const MY_CONSTANT = {}` I am telling you 'I will not mutate this, it is a constant'.

There should be no warning below?
<img width="1163" alt="image" src="https://github.com/ArnaudBarre/eslint-plugin-react-refresh/assets/17838632/711ca123-5ef3-49a4-9840-ee9d3f5c33eb">

**Comment by ArnaudBarre** [comment]
The issue is that you need also the React refresh runtime to skip that. I will probably add an option to the Vite React plugins to support Remix, and this option could be later used to support this kind of use case.

For now the safest way is to have a small file along the current file to export this constants

**Comment by DonnyVerduijn** [comment]
It seems, that this option also doesn't involve enums?

**Comment by ArnaudBarre** [comment]
Yeah TS enums are transpiled to JS objects so they are not "constants". I encourage you to use string based enums that are ignored because type only and aligned better with the current direction of TS to not add any runtime behaviour that is not defined in JS.

**Comment by ArnaudBarre** [comment]
Using enum is really something that goes against the current trend of TS to be just a typechecker for JS. These kind of runtime features (enum, namespaces, parameters properties) where all added a long ago.

I strongly advised using string enums:

```ts
export Type  = "a" |  "b";

interface ComponentProps {
  type: Type;
}

export function TypeComponent({ type }: ComponentProps) {
  return <div>{type}</div>;
}
```

```tsx
<TypeComponent type="a" />
```

**Comment by ArnaudBarre** [comment]
I'm doing some cleanup of the backlog, and few months later I don't think this worth adding. This would require a lot a coordinations with bundlers and still make the rule more confusing on what can be exported.
Primitive constant is good because it easy and ship to implement, but everything else then come with edge cases.

**Comment by githorse** [comment]
I also have a few constants that are objects or arrays (some basically enums, some not). To clarify, @ArnaudBarre - the current recommendation when using this plugin is to move those out of the component file into their own file, correct?

**Comment by ArnaudBarre** [comment]
Yep that's correct
